
FAIL TransformStream errors thrown in transform put the writable and readable in an errored state Can't find variable: TransformStream
FAIL TransformStream errors thrown in flush put the writable and readable in an errored state Can't find variable: TransformStream
FAIL errored TransformStream should not enqueue new chunks Can't find variable: TransformStream
FAIL TransformStream transformer.start() rejected promise should error the stream Can't find variable: TransformStream
FAIL when controller.error is followed by a rejection, the error reason should come from controller.error Can't find variable: TransformStream
FAIL TransformStream constructor should throw when start does assert_throws_js: constructor should throw function "() => new TransformStream({
    start() { throw new URIError('start thrown error'); },
    transform() {}
  })" threw object "ReferenceError: Can't find variable: TransformStream" ("ReferenceError") expected instance of function "function URIError() {
    [native code]
}" ("URIError")
FAIL when strategy.size throws inside start(), the constructor should throw the same error assert_throws_js: constructor should throw the same error strategy.size throws function "() => new TransformStream({
    start(c) {
      c.enqueue('a');
    },
    transform() {}
  }, undefined, strategy)" threw object "ReferenceError: Can't find variable: TransformStream" ("ReferenceError") expected instance of function "function URIError() {
    [native code]
}" ("URIError")
FAIL when strategy.size calls controller.error() then throws, the constructor should throw the first error assert_throws_js: the first error should be thrown function "() => new TransformStream({
    start(c) {
      controller = c;
      c.enqueue('a');
    },
    transform() {}
  }, undefined, strategy)" threw object "ReferenceError: Can't find variable: TransformStream" ("ReferenceError") expected instance of function "function URIError() {
    [native code]
}" ("URIError")
FAIL cancelling the readable side should error the writable Can't find variable: TransformStream
FAIL it should be possible to error the readable between close requested and complete Can't find variable: TransformStream
FAIL an exception from transform() should error the stream if terminate has been requested but not completed Can't find variable: TransformStream
FAIL abort should set the close reason for the writable when it happens before cancel during start, but cancel should still succeed Can't find variable: TransformStream
FAIL abort should set the close reason for the writable when it happens before cancel during underlying sink write, but cancel should still succeed Can't find variable: TransformStream
FAIL controller.error() should do nothing the second time it is called Can't find variable: TransformStream
FAIL controller.error() should do nothing after readable.cancel() Can't find variable: TransformStream
FAIL controller.error() should do nothing after writable.abort() has completed Can't find variable: TransformStream
FAIL controller.error() should do nothing after a transformer method has thrown an exception Can't find variable: TransformStream
FAIL erroring during write with backpressure should result in the write failing Can't find variable: TransformStream
FAIL a write() that was waiting for backpressure should reject if the writable is aborted Can't find variable: TransformStream
FAIL the readable should be errored with the reason passed to the writable abort() method Can't find variable: TransformStream

